include/deal.II/sundials/kinsol.h

```diff
@@ -0,0 +1,664 @@
+       *
+       * @param strategy Solution strategy
+       * @param maximum_non_linear_iterations Maximum number of nonlinear iterations
+       * @param function_tolerance Function norm stopping tolerance
+       * @param step_tolerance Scaled step stopping tolerance
+       *
+       * Newton parameters:
+       *
+       * @param no_init_setup No initial matrix setup
+       * @param maximum_setup_calls Maximum iterations without matrix setup
+       * @param maximum_newton_step Maximum allowable scaled length of the Newton step
+       * @param dq_relative_error Relative error for different quotient computation
+       *
+       * Linesearch parameters:
+       *
+       * @param maximum_beta_failures Maximum number of beta-condition failures
+       *
+       * Fixed point and Picard parameters:
+       *
+       * @param anderson_subspace_size Anderson acceleration subspace size
+       */
+      AdditionalData(
+        // Global parameters
+        const SolutionStrategy &strategy = linesearch,
+        const unsigned int &maximum_non_linear_iterations = 200,
+        const double &function_tolerance = 0.0,
+        const double &step_tolerance = 0.0,
+        const bool &no_init_setup = false,
+        const unsigned int &maximum_setup_calls = 0,
+        const double &maximum_newton_step = 0.0,
+        const double &dq_relative_error = 0.0,
+        const unsigned int &maximum_beta_failures = 0,
+        const unsigned int &anderson_subspace_size = 0) :
+        strategy(strategy),
+        maximum_non_linear_iterations(maximum_non_linear_iterations),
+        function_tolerance(function_tolerance),
+        step_tolerance(step_tolerance),
+        no_init_setup(no_init_setup),
+        maximum_setup_calls(maximum_setup_calls),
+        maximum_newton_step(maximum_newton_step),
+        dq_relative_error(dq_relative_error),
+        maximum_beta_failures(maximum_beta_failures),
+        anderson_subspace_size(anderson_subspace_size)
+      {};
+
+      /**
+       * Add all AdditionalData() parameters to the given ParameterHandler
+       * object. When the parameters are parsed from a file, the internal
+       * parameters are automatically updated.
+       *
+       * The following parameters are declared:
+       *
+       * @code
+       * @endcode
+       *
+       * These are one-to-one with the options you can pass at construction time.
+       *
+       * The options you pass at construction time are set as default values in
+       * the ParameterHandler object `prm`. You can later modify them by parsing
+       * a parameter file using `prm`. The values of the parameter will be updated
+       * whenever the content of `prm` is updated.
+       *
+       * Make sure that this class lives longer than `prm`. Undefined behaviour
+       * will occurr if you destroy this class, and then parse a parameter file
+       * using `prm`.
+       */
+      void add_parameters(ParameterHandler &prm)
+      {
```

Comment:
Move the implementation to a .cc file?

Reply:
I'm following the same conventions we have in the library for all other `AdditionalData` classes. If I move this to the `.cc` file, then I'd have to instantiate it for every vector type as well

Reply:
Since you're already instantiating the surrounding class, that is not much of an additional hurdle.

But let's not worry about this -- if it's that way everywhere else anyway, then let's just leave it as is.